### PR TITLE
[onert] Add unit tests for training ReLU op

### PR DESCRIPTION
**Pull request by ragmani**
This commit adds some tests to validate training for ReLU op.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
+
+    _context = std::make_unique<GenModelTrainContext>(cgen.finish());
+    _context->addTrainCase(
+      uniformTCD<float>({{{1, 3}}, {{2, 1}}},                                     // inputs
+                        {{{0, 1, 5, 5, 2, 1, 5, 5}}, {{2, 1, 5, 5, 0, 1, 5, 6}}}, // expected
+                        {13.5010f}                                                // loss
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc:52]: The result of executing the same model with the same data on tensorflow:
```
Epoch 1/5
2/2 [==============================] - 0s 4ms/step - loss: 13.9041 - categorical_accuracy: 0.0000e+00
Epoch 2/5
2/2 [==============================] - 0s 1ms/step - loss: 13.7684 - categorical_accuracy: 0.5000
Epoch 3/5
2/2 [==============================] - 0s 2ms/step - loss: 13.6340 - categorical_accuracy: 0.5000
Epoch 4/5
2/2 [==============================] - 0s 1ms/step - loss: 13.5010 - categorical_accuracy: 0.5000
Epoch 5/5
2/2 [==============================] - 0s 1ms/step - loss: 13.3693 - categorical_accuracy: 0.5000
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "GenModelTrain.h"
+
+TEST_F(GenModelTrain, NonTrainableOps_FC_Relu_FC)
+{
+  // (( Input 0 )) -> [ FC ] -> [ Relu ] -> [ FC ] -> (( Output 0 ))
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc:21]: If `(( Input 0 )) -> [ FC ] -> [ Relu ] -> (( Output 0 ))`, loss value does not changed. So I added one more FC layer.

**Comment by ragmani** [comment]
@nnfw-bot test onert-x64-release

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    cgen.addOperatorFullyConnected({{input, weight1, bias1}, {fc_output}});
+    cgen.addOperatorRelu({{fc_output}, {relu_output}});
+    cgen.addOperatorFullyConnected({{relu_output, weight2, bias2}, {output}});
+    cgen.setInputsAndOutputs({input}, {output});
+
+    float learning_rate = 0.01f;
+    int32_t batch_size = 1;
+    cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                       circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc:46]: If #13199 is merged first,
```suggestion
                       circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
    cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  // Invalid shape: output shape should be equal to input shape
+  int output = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_FLOAT32});
+  cgen.addOperatorRelu({{input}, {output}});
+  cgen.setInputsAndOutputs({input}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc:76]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  int input = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_INT32});
+  int output = cgen.addTensor({{2, 2}, circle::TensorType::TensorType_FLOAT32});
+  cgen.addOperatorRelu({{input}, {output}});
+  cgen.setInputsAndOutputs({input}, {output});
+
+  float learning_rate = 0.01f;
+  int32_t batch_size = 1;
+  cgen.addTrainInfo({circle::Optimizer::Optimizer_SGD, learning_rate,
+                     circle::LossFn::LossFn_MEAN_SQUARED_ERROR,
+                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
```
Comment (ragmani) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/Relu.test.cc:97]:
```suggestion
                     circle::LossReductionType::LossReductionType_SumOverBatchSize, batch_size});
  cgen.markAllOpsAsTrainable();
```

**Review by hseok-oh — APPROVED** [review]

**Review by jyoungyun — APPROVED** [review]
LGTM
